### PR TITLE
Feat/block exists

### PR DESCRIPTION
**Pull request by cmhulbert**

### CODE REVIEW

**Comment by axtimwalde** [comment]
Shouldn't this be part of `N5Reader` instead of `KeyValueAccess`?

**Comment by cmhulbert** [comment]
Yes, this PR adds a[ `blockExists` method to `N5Reader` ](https://github.com/saalfeldlab/n5/pull/104/files#diff-5d453aab496163573720c2bc998a6e31149431421efcff4b2a7d5e896c52aeeeR362)

File: src/main/java/org/janelia/saalfeldlab/n5/GsonKeyValueN5Reader.java [hunk 1]
```diff
@@ -61,6 +61,21 @@ default boolean datasetExists(final String pathName) throws N5Exception {
 		return getDatasetAttributes(pathName) != null;
 	}
 
+	@Override
+	default boolean blockExists(final String dataset, final long[] blockPosition) {
+		final DatasetAttributes datasetAttributes = getDatasetAttributes(dataset);
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/GsonKeyValueN5Reader.java:66]: Wondering if we should instead (or in addition) add
`blockExists(String name, DatasetAttributes datasetAttributes, long[] blockPosition`?

Comment (axtimwalde) [reply src/main/java/org/janelia/saalfeldlab/n5/GsonKeyValueN5Reader.java:66]: Reason: `getDatasetAttributes` incurs overhead that, in many practical cases, can be avoided?

Comment (cmhulbert) [reply src/main/java/org/janelia/saalfeldlab/n5/GsonKeyValueN5Reader.java:66]: I think that sounds like a good idea. We could potentially add `DatasetAttributes` to the cache too, but right now we don't (just cache the raw json tree)

**Pull request closed without merging**